atlasclient/utils.py

```diff
@@ -14,29 +14,19 @@
 
 """Utilities for the Python client for the MongoDB Atlas API."""
 
-import logging
-
-from http.client import HTTPConnection
-
 
 class JSONObject(dict):
     """Dictionary with dot-notation read access."""
+    def __coerce_dict(self, value):
+        if isinstance(value, dict):
+            return type(self)(value)
+        return value
```

Comment:
Does this effectively make nested dicts instead JSONObject immutable? Because we return a JSONObject copy of the actual value.

Reply:
It does indeed. I have now added a classmethod that casts all nested dicts as JSONObjects.